### DISCUSSION TITLE
Help with Programming problem(Beginner)!

### DISCUSSION THREAD

**Discussion by AurnobSYC** [community/community#21387] [Other Feature Feedback, Questions, & Ideas]
<p>I don’t know how to solve certain problems. How do i approach them? like the knights in a chessboard problem. Is there any theory that i need to study or do i have to find a pattern by trying different test cases? Thanks.</p>

**Comment by zencomplex** [comment, marked as answer]
<p>There are lots of books and resources out there that can help get you started. I recommend Problem Solving and Program Design in C – the 7th edition is searchable. This is a book used in college courses for computer science majors. C is a good language to start with, in my opinion, because it has been around forever and it’s a pretty “basic” language.</p>
<p>There’s a chapter in the book that was extremely helpful to me in providing methods to breaking down a problem into its parts (chapter 3.3), but I’d recommend that you start at the very beginning and go from there, even with some of the content that may not seem like it’s relevant to you now, because you really never know when you’re going to need it.</p>

**Comment by AurnobSYC** [comment]
<p>I know c. now studying c++ for my next course. I know the basics i think. Im having problems with competitive problems. Not sure if i need to study theories for them or have to think. Like the problem lightoj 1010. Is there any theory to solve such problems? Or do i have to find the pattern by trying different test case results by hand and find the pattern. </p>

**Comment by badislezghad** [comment]
<p>vielen dank fur deine unterstutzung und hilfe ich kann dir sagen dass informationen sind genug und richtig                        ich habe viel probleme in programming ich frage dich ob hast einige bucher in der domain oder nicht weil diese bucher hilfen mir sehr danke schone</p>
